maker/src/position.rs

```diff
@@ -0,0 +1,220 @@
+use async_trait::async_trait;
+use rust_decimal::Decimal;
+use std::collections::HashSet;
+use std::fmt::Display;
+use std::hash::Hash;
+use std::ops::Add;
+use uuid::Uuid;
+
+mod tentenone;
+
+pub struct Manager {
+    position: Position,
+}
+
+impl Manager {
+    pub fn new() -> Self {
+        Self {
+            position: Position::new(),
+        }
+    }
+}
+
+impl Default for Manager {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+pub struct PositionUpdateTenTenOne(pub Vec<OrderTenTenOne>);
+
+#[derive(Debug)]
+pub struct OrderTenTenOne {
+    order_id: Uuid,
+    contract_symbol: ContractSymbol,
+    /// The number of contracts corresponding to this position update.
+    ///
+    /// The sign determines the direction: positive is long; negative is short.
+    contracts: Decimal,
+}
+
+impl OrderTenTenOne {
+    pub fn new(order_id: Uuid, contract_symbol: ContractSymbol, contracts: Decimal) -> Self {
+        Self {
+            order_id,
+            contract_symbol,
+            contracts,
+        }
+    }
+}
+
+#[async_trait]
+impl xtra::Actor for Manager {
+    type Stop = ();
+
+    async fn stopped(self) -> Self::Stop {}
+}
+
+#[async_trait]
+impl xtra::Handler<PositionUpdateTenTenOne> for Manager {
+    type Return = ();
+
+    async fn handle(
+        &mut self,
+        update: PositionUpdateTenTenOne,
+        _: &mut xtra::Context<Self>,
+    ) -> Self::Return {
+        for OrderTenTenOne {
+            order_id,
+            contract_symbol,
+            contracts,
+        } in update.0
+        {
+            self.position
+                .update_tentenone(contract_symbol, order_id, contracts)
+        }
+    }
+}
+
+/// The overall position of the maker.
+#[derive(Debug)]
+struct Position {
+    tentenone: HashSet<tentenone::Position>,
+}
+
+impl Position {
+    pub fn new() -> Self {
+        Self {
+            tentenone: HashSet::from_iter([tentenone::Position::new(ContractSymbol::BtcUsd)]),
+        }
+    }
+
+    fn update_tentenone(
+        &mut self,
+        contract_symbol: ContractSymbol,
+        order_id: Uuid,
+        contracts: Decimal,
+    ) {
+        let mut position = self
+            .tentenone
+            .get(&contract_symbol)
+            .cloned()
+            .unwrap_or(tentenone::Position::new(ContractSymbol::BtcUsd));
+
+        position.update(order_id, contracts);
+
+        self.tentenone.replace(position);
+    }
+
+    #[cfg(test)]
+    fn get_tentenone(&self, contract_symbol: &ContractSymbol) -> Decimal {
```

Comment:
Should we add an HTTP Api to query this ? Could come handy in our production setup to know what is going on .